### PR TITLE
chore(root): Move @novu/application generic under libs

### PR DESCRIPTION
**Pull request by SokratisVidros**
### What changed? Why was the change needed?
This package contains shared code for Novu services. It is considered an internal library and should be deprecated from NPM as it can't be used independently from Novu services powered by Nest.js.

### CODE REVIEW

File: libs/application-generic/package.json [hunk 1]
```diff
@@ -5,11 +5,7 @@
   "main": "build/main/index.js",
   "typings": "build/main/index.d.ts",
   "module": "build/module/index.js",
-  "private": false,
-  "repository": {
-    "url": "https://github.com/novuhq/novu",
-    "directory": "packages/nest"
-  },
+  "private": true,
```
Comment (SokratisVidros) [inline libs/application-generic/package.json:8]: All the other files are moved under the `libs` folder. Package.json was changed to make this package private.

File: libs/application-generic/package.json [hunk 2]
```diff
@@ -138,18 +130,6 @@
     "LICENSE",
     "README.md"
   ],
-  "ava": {
```
Comment (SokratisVidros) [inline libs/application-generic/package.json:141]: I guess we are not using ava anymore :)

Comment (rifont) [reply libs/application-generic/package.json:141]: 🧹

File: libs/application-generic/src/usecases/workflow/update-workflow/update-workflow.command.ts [hunk 3]
```diff
@@ -16,7 +16,7 @@ import {
 } from '@novu/shared';
 
 import { EnvironmentWithUserCommand } from '../../../commands';
-import { NotificationStep } from '../../../usecases';
+import { NotificationStep } from '../..';
```
Comment (rifont) [inline libs/application-generic/src/usecases/workflow/update-workflow/update-workflow.command.ts:19]:
```suggestion
import { NotificationStep } from '../../usecases';
```

I often find these implicit imports from the index file to be troublesome for introduction of cyclic dependencies. Can we revert?

File: libs/application-generic/src/usecases/workflow/update-workflow/update-workflow.usecase.ts [hunk 4]
```diff
@@ -37,7 +37,7 @@ import {
   CreateMessageTemplateCommand,
   NotificationStep,
   NotificationStepVariantCommand,
-} from '../../../usecases';
+} from '../..';
```
Comment (rifont) [inline libs/application-generic/src/usecases/workflow/update-workflow/update-workflow.usecase.ts:40]: Same as above, risk of cyclic dependency
```suggestion
} from '../../usecases';
```

**Review by rifont — APPROVED** [review]
Awesome! Nice tidy up too.

**Comment by rifont** [comment]
The PR appears to need another install, the [Get Affected Packages](https://github.com/novuhq/novu/actions/runs/9165935028/job/25200346545?pr=5612#step:5:183) workflow is complaining about lockfile changes.

Got a few pesky cspell [errors](https://github.com/novuhq/novu/actions/runs/9165935028/job/25200346827?pr=5612) too.

**Comment by SokratisVidros** [comment]
> The PR appears to need another install, the [Get Affected Packages](https://github.com/novuhq/novu/actions/runs/9165935028/job/25200346545?pr=5612#step:5:183) workflow is complaining about lockfile changes.
> 
> Got a few pesky cspell [errors](https://github.com/novuhq/novu/actions/runs/9165935028/job/25200346827?pr=5612) too.

👍  The first one with the lockfile is trickier than it looks as all the pnpm-lock changes are committed but pnpm still refuses to cooperate. The good news is that it's reproducible locally so I will spend some time to figure out it today.

**Pull request merged**